src/solvers/mp/rcspp/RCGraph.cpp

```diff
@@ -0,0 +1,1095 @@
+  }
+  for(int a=0; a<nArcs_; a++) nArcsPerType.at(arcType(a))++;
+  // DISPLAY ARCS
+  rep << "#     -------------------------" << std::endl;
+  rep << "#   > ARCS                     " << std::endl;
+  rep << "#     -------------------------" << std::endl;
+  for(int t = SOURCE_TO_PRINCIPAL; t!=NONE_ARC; t++){
+    ArcType ty = static_cast<ArcType>(t);
+    rep << "#        " << arcTypeName[ty] << "  " <<nArcsPerType.at(ty) << std::endl;
+  }
+  rep << "#     -------------------------" << std::endl;
+  rep << "#        TOTAL            " << nArcs_ << std::endl;
+  rep << "#     -------------------------" << std::endl;
+
+  return rep.str();
+}
+
+// Print the path (arcs, nodes, cost of each arc in the current network, etc.)
+//
+void RCGraph::printPath(std::vector< boost::graph_traits<Graph>::edge_descriptor > path, spp_res_cont resource) const {
+
+  // The successive nodes, and corresponding arc costs / time
+  //
+  std::cout << "# " << std::endl;
+  for( int j = static_cast<int>( path.size() ) - 1; j >= 0;	--j){
+    int a = boost::get(&Arc_Properties::num, g_, path[j]);
+    std::cout << "# \t| [ " << shortNameNode(boost::source( path[j], g_ )) << " ]";
+    std::cout << "\t\tCost:  " << arcCost(a);
+    const std::vector<int> & consumptions = arcConsumptions(a);
+    for(int l=0; l<consumptions.size(); ++l)
+      std::cout << "\t\t" << labelName[l] << ":" << consumptions[l];
+    std::cout << "\t\t[" << (arcForbidden(a) ? "forbidden" : " allowed ") << "]" << std::endl;
+  }
+  std::cout << "# " << std::endl;
+  for( int j = static_cast<int>( path.size() ) - 1; j >= 0;	--j){
+    int a = boost::get(&Arc_Properties::num, g_, path[j]);
+    std::cout << printArc(a) << std::endl;
+  }
+
+  // Last node and total
+  //
+  std::cout << "# \t| [";
+  for(int s: sinks_) std::cout << shortNameNode(s) << " ";
+  std::cout << "]" << std::endl;
+  std::cout << "# \t| ~TOTAL~   \t\tCost:   " << resource.cost;
+  for(int l=0; l<resource.size(); ++l)
+    std::cout << "\t\t" << labelName[l] << ":" << resource.label_value(l);
+  std::cout << std::endl << "# \t| " << std::endl;
+  std::cout << "# \t| Rotation: |";
+
+  // Print it "as a rotation"
+  //
+  int k=0;
+  int firstDay = -1;
+  for( int j = static_cast<int>( path.size() ) - 1; j >= 0;	--j){
+    if(firstDay == -1 && boost::source( path[j], g_ ) == source_) {
+      firstDay = boost::get(&Arc_Properties::day, g_, path[j]);
+      while (k < firstDay) {
+        std::cout << " |";
+        k++;
+      }
+    }
+    for(int s: get( &Arc_Properties::shifts, g_, path[j])) {
+      std::cout << s << "|";
+      k++;
+    }
+  }
+  while(k < nDays_){
+    std::cout << " |";
+    k++;
+  }
+  std::cout << std::endl;
+}
+
+/*************************************************************************
+ * A GARDER AU CAS OU COMME EXEMPLE POUR CERTAINES FONCTIONS / SYNTAXES. *
+ *************************************************************************/
+enum nodes { A, B, C, D, E, F, G };
```

Comment:
Test functions should not be in the same file as other functions. Use to create unitary test ? (if yes, create issue I guess)
Delete otherwise...

Reply:
I've removed it  on next version